### ISSUE TITLE
Upgrade esprima, estraverse, and esutils dependencies

### ISSUE THREAD

**Issue by stuhlmueller** [probmods/webppl#69]
We're using 1.x.x versions; the projects are at 2.x.x or 3.x.x by now. There are probably some backwards-incompatible changes, so upgrading will take a bit of adjustment.

**Comment by pedir** [comment]
I upgraded and tested most of our dependancies to the their latest versions. The only issue is from upgrading estraverse from version 2.0.0 to 3.0.0. I narrowed down this issue and it's specifically caused by change in this line in estraverse which is used in our syntax.js file (```var keys = require('estraverse').VisitorKeys;```):

Version 2.0.0 (which works):
```FunctionExpression: ['id', 'params', 'defaults', 'rest', 'body'],```

Version 3.0.0 (does't work):
```FunctionExpression: ['id', 'params', 'body'],```

I'm still not that familiar with the specifics of our code transformation, but a fix should be easy and I'll try to figure it out.
